src/AppBundle/Service/FileUploader.php

```diff
@@ -49,23 +54,44 @@ public function uploadLogo(Request $request)
     }
 
     /**
+     * @param Request $request
+     *
+     * @return string
+     */
+    public function uploadReceipt(Request $request)
+    {
+        $file = $this->getFileFromRequest($request);
+
+        $mimeType = $file->getMimeType();
+        $fileType = explode('/', $mimeType)[0];
+        if ($fileType === 'image') {
+            return $this->uploadFile($file, $this->receiptFolder);
+        } else {
+            throw new BadRequestHttpException('Filtypen må være et bilde.');
+        }
+    }
+
+    /**
      * @param UploadedFile $file
      * @param string       $targetFolder
      *
      * @return string absolute file path
      */
     public function uploadFile(UploadedFile $file, string $targetFolder)
     {
-        $originalFileName = $file->getClientOriginalName();
         $fileExt = $file->guessExtension();
-
         $fileName = $this->generateRandomFileNameWithExtension($fileExt);
 
-        $relativePath = $this->getRelativePath($targetFolder, $fileName);
+        if (!is_dir($targetFolder)) {
+            mkdir($targetFolder, 0777, true); // Should mode be 777? 664 maybe? 644?
+        }
 
-        $uploadSuccessful = move_uploaded_file($file->getPathname(), $relativePath);
+        try {
+            $file->move($targetFolder, $fileName);
```

Comment:
`move_uploaded_file` fungerer ikke i phpunit nemlig. Men `UploadedFile::move` sjekker `$test` parameteret til `UploadedFile`en, og bestemmer ut i fra det om den skal bruke `File::move_uploaded_file` eller `File::move`. Slik jeg har forstått det er det altså nødvendig for å tillate filopplastninger i tester.